src/test/java/org/owasp/esapi/reference/DefaultSecurityConfigurationTest.java

```diff
@@ -18,467 +18,475 @@
+        assertEquals(Logger.WARNING, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "trace");
+        assertEquals(Logger.TRACE, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "Off");
+        assertEquals(Logger.OFF, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "all");
+        assertEquals(Logger.ALL, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "DEBUG");
+        assertEquals(Logger.DEBUG, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "info");
+        assertEquals(Logger.INFO, secConf.getLogLevel());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_LEVEL, "ERROR");
+        assertEquals(Logger.ERROR, secConf.getLogLevel());
+    }
+    
+    @Test
+    public void testGetLogFileName() {
+        DefaultSecurityConfiguration secConf = new DefaultSecurityConfiguration(new java.util.Properties());
+        assertEquals("ESAPI_logging_file", secConf.getLogFileName());
+        
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.LOG_FILE_NAME, "log.txt");
+        assertEquals("log.txt", secConf.getLogFileName());
+    }
+    
+    @Test
+    public void testGetMaxLogFileSize() {
+        DefaultSecurityConfiguration secConf = new DefaultSecurityConfiguration(new java.util.Properties());
+        assertEquals(DefaultSecurityConfiguration.DEFAULT_MAX_LOG_FILE_SIZE, secConf.getMaxLogFileSize());
+        
+        int maxLogSize = (1024 * 1000);
+        secConf = this.createWithProperty(DefaultSecurityConfiguration.MAX_LOG_FILE_SIZE, String.valueOf(maxLogSize));
+        assertEquals(maxLogSize, secConf.getMaxLogFileSize());
+    }
+    
+    private String patternOrNull(Pattern p){
+        return null==p?null:p.pattern();
+    }
 
-	@Test
-	public void testValidationsPropertiesFileOptions(){
-		DefaultSecurityConfiguration secConf = new DefaultSecurityConfiguration("ESAPI-SingleValidatorFileChecker.properties");
-		assertEquals(patternOrNull(secConf.getValidationPattern("Test1")), "ValueFromFile1");
-		assertNull(secConf.getValidationPattern("Test2"));
-		assertNull(secConf.getValidationPattern("TestC"));
-		
-		secConf = new DefaultSecurityConfiguration("ESAPI-DualValidatorFileChecker.properties");
-		assertEquals(patternOrNull(secConf.getValidationPattern("Test1")), "ValueFromFile1");
-		assertEquals(patternOrNull(secConf.getValidationPattern("Test2")), "ValueFromFile2");
-		assertNull(secConf.getValidationPattern("TestC"));
+    @Test
+    public void testRootCPLoading(){
+        DefaultSecurityConfiguration secConf = new DefaultSecurityConfiguration("ESAPI-root-cp.properties");
```

Comment:
@JoergAdler Rather creating yet another variant of **ESAPI.properties** under "**src/test/resources/esapi**", could you please change your test to use one of these:

```
src/test/resources/esapi/ESAPI-DualValidatorFileChecker.properties
src/test/resources/esapi/ESAPI.properties
src/test/resources/esapi/ESAPI-SingleValidatorFileChecker.properties
src/test/resources/esapi/ESAPI-QuotedValidatorFileChecker.properties
src/test/resources/esapi/ESAPI-test-2.properties
src/test/resources/esapi/ESAPI-CommaValidatorFileChecker.properties
src/test/resources/esapi/ESAPI-test.properties
```
instead? There are already way too many of these and every time we add a new property to ESAPI or delete an old property, I feel compelled to keep all of these others in sync with the main one in `configuration/esapi/ESAPI.properties`, (and admittedly, have been doing a poor job) in part because we have been unable to figure out how to get the stuff under 'configuration/' into the 'sources' jar so people keep grabbing on of the test ones from `src/est/resources/esapi` and using those for production. (Okay; we do know how to get it into the sources jar; we just haven't figured out how to do that with also getting it built into the regular **esapi-**_release_**.jar**.)

I am fine with adding a custom property or two to one of these others. I'm _not_ fine with creating yet another **ESAPI.properties** file under `src/test/resources/esapi`.

Reply:
Hi. 

The point of this test is to make sure, that an esapi-properties from the root of the classpath (cp) is loaded. 
So it has to be in src/test/resources/ and has to have a different name than the other ones, to prove, if not one of the other files is loaded accidentally.

Reply:
I was looking at this a little bit, and I don't see a better way to prove the case being targeted by this change.   I will try to do a little more independent verification on it, but I think this change is a good thing.

@kwwall  if we could ensure that all copies of the ESAPI.properties in test scope emulate the values in the src/main directory would that alleviate some concern?

There may be some magic we can do with maven resource copy to duplicate the ESAPI.properties file into test scope with new names, and possibly new values for certain keys.  It's been a little while since I looked at it, but I believe this is a feasible thing.

If we did that then the files would stay in-sync with the primary properties.  In addition, only the src/main version would be in source control.  All others would be created during mvn package for the purpose of unit testing. We may even be able to explicitly add all src/test/**/ESAPI*.properties to the .gitignore file at that point so they don't get committed back up.

Aside from changing how the files are managed or updating the specific packaging of the resources in the delivery I'm not sure there's another way to address it.  Both of those efforts feel larger than this review though, so I hesitate to say that this change requires either of those things to be done first.  Especially when this is consistent with the current pattern and we are aware of the issue and are working to address it.

---
The one thing that may work (which I haven't tested) would be put this file with the other test resource files.  In the test use the beforeAll to copy the resource content to the root of the classpath; likewise, the afterall must remove the created file.
This feels heavy-handed to me, but may be another feasible interim solution while the larger resource management problems are being addressed.

Reply:
Certainly we could less the possible confusion (and my objection!) if we strip the new 'ESAPI-root-cp.properties' to the bare bones properties of what is actually needed here. I'm pretty sure that we don't need anything BUT
`ESAPI.Logger=org.owasp.esapi.reference.Log4JLogFactory`
and maybe the rest of the 'Logger.*' property name / value pairs. But other than the ESAPI logger properties, I don't think anything else should be required. If @JoergAdler agrees to do that, then I suppose I can live with the results.